FirebaseDynamicLinks/Sources/FIRDynamicLinkNetworking.m

```diff
@@ -116,33 +117,66 @@ - (void)resolveShortLink:(NSURL *)url
     @"sdk_version" : FDLSDKVersion
   };
 
-  FIRNetworkRequestCompletionHandler resolveLinkCallback = ^(NSData *data, NSError *error) {
-    NSURL *resolvedURL;
-
-    if (!error && data) {
-      NSDictionary *result = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
-      if ([result isKindOfClass:[NSDictionary class]]) {
-        id invitationIDObject = [result objectForKey:@"invitationId"];
-
-        NSString *invitationIDString;
-        if ([invitationIDObject isKindOfClass:[NSDictionary class]]) {
-          NSDictionary *invitationIDDictionary = invitationIDObject;
-          invitationIDString = invitationIDDictionary[@"id"];
-        } else if ([invitationIDObject isKindOfClass:[NSString class]]) {
-          invitationIDString = invitationIDObject;
+  FIRNetworkRequestCompletionHandler resolveLinkCallback = ^(NSData *data, NSURLResponse *response,
+                                                             NSError *error) {
+    NSURL *resolvedURL = nil;
+    NSHTTPURLResponse *resp = nil;
+
+    if (![response isKindOfClass:[NSHTTPURLResponse class]]) {
+      NSError *err = [NSError
+          errorWithDomain:kGenericErrorDomain
+                     code:0
+                 userInfo:@{@"message" : @"Response should be of type NSHTTPURLResponse."}];
+      handler(resolvedURL, error ? error : err);
```

Comment:
Curious, what is the context here in deciding which error to pass in the completion? 

Also, nit: perhaps rename `err` to `responseTypeError`, `customError`, or something like that

Reply:
Per `FIRNetworkRequestCompletionHandler` definition `error` object is nullable. I decided to give precedence to the original error over the local one (`err`), because I didn't want to shadow potentially more serious issue, e.g. stemming from server side, given that initially `response` object had been disregarded completely.

Reply:
Renamed `err` -> `responseTypeError`.